crates/client-api-messages/src/timestamp.rs

```diff
@@ -1,30 +1,33 @@
 use std::time::{Duration, SystemTime};
 
-use spacetimedb_sats::{impl_deserialize, impl_serialize};
+use spacetimedb_sats::SpacetimeType;
 
-#[derive(Copy, Clone, PartialEq, Eq, Debug, serde::Serialize)]
-#[serde(transparent)]
-#[repr(transparent)]
-pub struct Timestamp(pub u64);
+#[derive(SpacetimeType, Copy, Clone, PartialEq, Eq, Debug, serde::Serialize)]
```

Comment:
Yes, it will. Unfortunately, `transparent` newtypes don't really work for codegen, so we have to use actual single-element products if we want to generate a named type. I discussed how we might implement them in C# with Ingvar, and the conclusion was that there isn't a good way.